Spigot-Server-Patches/0581-Add-tps-to-gui.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: AlexProgrammerDE <40795980+AlexProgrammerDE@users.noreply.github.com>
+Date: Fri, 2 Oct 2020 19:00:50 +0200
+Subject: [PATCH] Add tps to gui
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/gui/RAMDetails.java b/src/main/java/com/destroystokyo/paper/gui/RAMDetails.java
+index e463a86a6ad6c4f2885f4df159411bad294e1208..c3633c9806f5bfd1caae52636c275e1063e01f2a 100644
+--- a/src/main/java/com/destroystokyo/paper/gui/RAMDetails.java
++++ b/src/main/java/com/destroystokyo/paper/gui/RAMDetails.java
+@@ -58,9 +58,15 @@ public class RAMDetails extends JList<String> {
+     public void update() {
+         GraphData data = RAMGraph.DATA.peekLast();
+         Vector<String> vector = new Vector<>();
++
++        // Paper start - Add tps entry
++        double tps = Bukkit.getTPS()[0];
+         vector.add("Memory use: " + (data.getUsedMem() / 1024L / 1024L) + " mb (" + (data.getFree() * 100L / data.getMax()) + "% free)");
+         vector.add("Heap: " + (data.getTotal() / 1024L / 1024L) + " / " + (data.getMax() / 1024L / 1024L) + " mb");
+         vector.add("Avg tick: " + DECIMAL_FORMAT.format(getAverage(server.getTickTimes())) + " ms");
++        vector.add("TPS: " + ( ( tps > 21.0 ) ? "*" : "" ) + Math.min( Math.round( tps * 100.0 ) / 100.0, 20.0 ));
```

Comment:
Perhaps rather print it like with /tps?

Reply:
ok, but i thought it might look a little bit too large for smaller scaled windows

Reply:
also that code is actually from the tps command. I just didn't really want to make it look too large